src/dmd/dtemplate.d

```diff
@@ -476,6 +476,17 @@ extern (C++) final class Tuple : RootObject
 {
     Objects objects;
 
+    /**
+     *  Constructs a new instance.
+     *  Params:
+     *      numObjects = The initial number of objects.
+     */
+    extern (D) this(size_t numObjects = 0)
```

Comment:
See #12 here:  https://forum.dlang.org/post/p6oibo$1lmi$1@digitalmars.com

I don't really like it, but you can avoid the default parameter by forwarding to a parameter-less constructor in the body of this constructor.

Reply:
Yes but it's either that or a useless default ctor since default ctors ares till not auto generated ;)

Reply:
I'm guessing the preference would be:

```D
/**
Constructs a new instance.
*/
extern (D) this()
{
    this(0);
}

/**
Constructs a new instance.

Params:
    numObjects = The initial number of objects.
*/
extern (D) this(size_t numObjects)
{
    objects.setDim(numObjects);
}
```